governance/contract.go

```diff
@@ -96,11 +98,25 @@ func (e *ContractEngine) contractGetAllParamsAt(num uint64) (*params.GovParamSet
 		return params.NewGovParamSet(), nil
 	}
 
-	caller := &contractCaller{
-		chain:        chain,
-		contractAddr: addr,
+	caller := backends.NewBlockchainContractCaller(chain)
+	contract, _ := govcontract.NewGovParamCaller(addr, caller)
+
+	names, values, err := contract.GetAllParamsAt(nil, new(big.Int).SetUint64(num))
+	if err != nil {
+		logger.Warn("ContractEngine disabled: getAllParams call failed", "err", err)
+		return params.NewGovParamSet(), nil
+	}
+
+	if len(names) != len(values) {
```

Comment:
Is it possible to have a name but no value?

Reply:
It is impossible unless the GovParam contract is broken. If somehow the GovParam contract is broken, this safety check would prevent OOB in the following code.

We could try to panic without this check, however it would be impossible to fix the broken GovParam contract when the node panics and stops proposing blocks.